core/src/main/java/io/questdb/griffin/engine/functions/catalogue/WalTableListFunctionFactory.java

```diff
@@ -0,0 +1,240 @@
+                        }
+                    }
+                    boolean isDir = Files.isDir(ff.findName(findPtr), ff.findType(findPtr), tableNameSink);
+                    if (isDir) {
+                        boolean isWalTable = TableSequencerAPI.isWalTable(tableNameSink, rootPath, ff);
+                        rootPath.trimTo(rootLen);
+                        if (isWalTable) {
+                            if (record.switchTo(tableNameSink)) {
+                                return true;
+                            }
+                        }
+                    }
+                }
+            }
+
+            @Override
+            public void recordAt(Record record, long atRowId) {
+                throw new UnsupportedOperationException();
+            }
+
+            @Override
+            public long size() {
+                return -1;
+            }
+
+            @Override
+            public void toTop() {
+                close();
+            }
+
+            public class TableListRecord implements Record {
+                private long sequencerTxn;
+                private boolean suspendedFlag;
+                private long writerTxn;
+
+                @Override
+                public boolean getBool(int col) {
+                    if (col == suspendedColumn) {
+                        return suspendedFlag;
+                    }
+                    return false;
+                }
+
+                @Override
+                public long getLong(int col) {
+                    if (col == writerTxnColumn) {
+                        return writerTxn;
+                    }
+                    if (col == sequencerTxnColumn) {
+                        return sequencerTxn;
+                    }
+                    return Numbers.LONG_NaN;
+                }
+
+                @Override
+                public CharSequence getStr(int col) {
+                    if (col == nameColumn) {
+                        return tableNameSink;
+                    }
+                    return null;
+                }
+
+                @Override
+                public CharSequence getStrB(int col) {
+                    return getStr(col);
+                }
+
+                @Override
+                public int getStrLen(int col) {
+                    return getStr(col).length();
+                }
+
+                public boolean switchTo(final CharSequence tableName) {
+                    final CairoEngine cairoEngine = sqlExecutionContext.getCairoEngine();
+                    final TableSequencerAPI sequencerAPI = cairoEngine.getTableSequencerAPI();
+
+                    suspendedFlag = sequencerAPI.isSuspended(tableName);
+                    sequencerTxn = sequencerAPI.lastTxn(tableName);
+                    writerTxn = -1;
+                    try (TableReader reader = cairoEngine.getReader(sqlExecutionContext.getCairoSecurityContext(), tableName)) {
+                        writerTxn = reader.getTxn();
```

Comment:
writerTxn should be taken from `TxReader.seqTxn`

Reply:
Fixed